### ISSUE TITLE
Change request to improve FOR statement structure

### ISSUE THREAD

**Issue by PauliusKu** [usagi-coffee/tree-sitter-abl#58]
Currently we have this FOR statement structure:

```
for_statement
--- EACH
--- identifier
--- where_clause
...
--- body
------ abl_statement
...

```

We ran into problems when trying to implement code formatting logic for this structure. The main issue is that header nodes are under the same level as `body` node. As `body` starts with the first statement inside it, and not with `:` symbol, for formatting proposes we have to use whole range of `for_statement` node. And this leads to further issues when trying to create formatting algorithm.

The best solution for us would be to have additional node (e.g. `for_header`). Something like this:

```
for_statement
--- for_header
------ EACH
------ identifier
------ where_clause
...
--- body
------ abl_statement
...

```

**Comment by jkbz64** [comment]
Would reworking body element to include ':' character instead work for you?

**Comment by PauliusKu** [comment]
Yes, that would be fine.

**Comment by jkbz64** [comment]
Should be fixed with b1f5901
